Rework learning goals and make them consistent (#190)
Simulation-Software-Engineering/Lecture-Material: 03_building_and_packaging/README.md

Learning goals:

- Explain why software is packaged.

> - Create Python packages, upload to PyPI, and install with pip.

Reviewer: ```suggestion
- Create Python packages, publish on PyPI, and install with pip.
```

- Understand the difference between static and dynamic libraries and common ways of installation on Linux.
- Build C++ software and handle dependencies with Make and CMake.
- Package C++ software with CPack and create Debian packages.

> - Create Spack packages for high-performance computing.

Reviewer: Is the focus really on HPC here? Are we looking into optimization options, for example?

Author: No, we don't IIRC. We introduce Spack as an alternative if you are on a cluster (without admin rights).

Reviewer: ```suggestion
- Create Spack packages, e.g., for high-performance computing systems with restricted access rights.
```
